ddtrace/internal/_queue.pyx

```diff
@@ -0,0 +1,52 @@
+import threading
+
+from ddtrace.vendor import attr
+from ..internal.logger import get_logger
+from . import _rand
+
+
+log = get_logger(__name__)
+
+
+@attr.s
+class TraceQueue(object):
+
+    maxsize = attr.ib(type=int, default=0)
+    _lock = attr.ib(init=False, factory=threading.Lock)
+    _queue = attr.ib(init=False, factory=list)
+    _accepted = attr.ib(init=False, type=int, default=0)
+    _accepted_lengths = attr.ib(init=False, type=int, default=0)
+    _dropped = attr.ib(init=False, type=int, default=0)
+
+    def __len__(self):
+        return len(self._queue)
+
+    def put(self, item):
+        with self._lock:
+            if self.maxsize <= 0 or len(self._queue) < self.maxsize:
+                self._queue.append(item)
+            else:
+                idx = _rand.rand64bits() % len(self._queue)
+                self._queue[idx] = item
+
+                log.warning("Writer queue is full has more than %d traces, dropping trace", self.maxsize)
```

Comment:
```suggestion
                log.warning("Writer queue is full has more than %d traces, dropping a random trace", self.maxsize)
```

Not sure it's important, but the current wording makes me think that the latest trace is dropped.

Reply:
The whole message is weirdly worded haha. Fixing.

Reply:
I must have written this 😆